jax_distribution/systems/ippo_ff_anakin.py

```diff
@@ -0,0 +1,475 @@
+import optax
+from flax import struct
+from flax.linen.initializers import constant, orthogonal
+from jumanji.environments.routing.multi_cvrp.generator import UniformRandomGenerator
+from jumanji.environments.routing.multi_cvrp.types import State
+from jumanji.types import TimeStep
+from jumanji.wrappers import Wrapper
+
+
+class TimeIt:
+    def __init__(self, tag, frames=None):
+        self.tag = tag
+        self.frames = frames
+
+    def __enter__(self):
+        self.start = timeit.default_timer()
+        return self
+
+    def __exit__(self, *args):
+        self.elapsed_secs = timeit.default_timer() - self.start
+        msg = self.tag + (": Elapsed time=%.2fs" % self.elapsed_secs)
+        if self.frames:
+            msg += ", FPS=%.2e" % (self.frames / self.elapsed_secs)
+        print(msg)
+
+
+@struct.dataclass
+class LogEnvState:
+    env_state: State
+    episode_returns: float
+    episode_lengths: int
+    returned_episode_returns: float
+    returned_episode_lengths: int
+
+
+class LogWrapper(Wrapper):
+    """Log the episode returns and lengths."""
+
+    def reset(self, key: chex.PRNGKey) -> Tuple[LogEnvState, TimeStep]:
+        state, timestep = self._env.reset(key)
+        # timestep.extras = {}
+        state = LogEnvState(state, 0.0, 0, 0.0, 0)
+        return state, timestep
+
+    def step(
+        self,
+        state: State,
+        action: jnp.ndarray,
+    ) -> Tuple[State, TimeStep]:
+
+        env_state, timestep = self._env.step(state.env_state, action)
+
+        new_episode_return = state.episode_returns + jnp.mean(timestep.reward)
+        new_episode_length = state.episode_lengths + 1
+        state = LogEnvState(
+            env_state=env_state,
+            episode_returns=new_episode_return * (1 - timestep.last()),
+            episode_lengths=new_episode_length * (1 - timestep.last()),
+            returned_episode_returns=state.returned_episode_returns
+            * (1 - timestep.last())
+            + new_episode_return * timestep.last(),
+            returned_episode_lengths=state.returned_episode_lengths
+            * (1 - timestep.last())
+            + new_episode_length * timestep.last(),
+        )
+        return state, timestep
+
+
+class Transition(NamedTuple):
+    done: jnp.ndarray
+    action: jnp.ndarray
+    value: jnp.ndarray
+    reward: jnp.ndarray
+    log_prob: jnp.ndarray
+    observation: jnp.ndarray
+    info: jnp.ndarray
+
+
+class ActorCritic(nn.Module):
+    action_dim: Sequence[int]
+    activation: str = "tanh"
+    env_name: str = "RobotWarehouse-v0"
+
+    @nn.compact
+    def __call__(self, observation):
+        if "MultiCVRP" in self.env_name:
+            x = observation.vehicles.coordinates  # .shape
+        elif "RobotWarehouse" in self.env_name:
+            x = observation.agents_view
+
+        if self.activation == "relu":
+            activation = nn.relu
+        else:
+            activation = nn.tanh
+
+        actor_mean = nn.Dense(
```

Comment:
Maybe we can replace `actor_mean` with a slightly more descriptive name? Like actor output?